src/main/java/li/cil/scannable/common/Scannable.java

```diff
@@ -1,38 +1,25 @@
 package li.cil.scannable.common;
 
-import li.cil.scannable.api.API;
-import li.cil.scannable.client.ClientSetup;
-import li.cil.scannable.client.scanning.ScanResultProviders;
 import li.cil.scannable.client.ClientConfig;
 import li.cil.scannable.common.config.CommonConfig;
 import li.cil.scannable.common.container.Containers;
 import li.cil.scannable.common.item.Items;
+import li.cil.scannable.common.network.Network;
 import li.cil.scannable.common.tags.ItemTags;
 import li.cil.scannable.util.ConfigManager;
-import li.cil.scannable.util.RegistryUtils;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.fml.DistExecutor;
-import net.minecraftforge.fml.common.Mod;
-import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.fabricmc.api.ModInitializer;
 
-@Mod(API.MOD_ID)
-public final class Scannable {
-    public Scannable() {
+public final class Scannable implements ModInitializer {
+    public void onInitialize() {
         ConfigManager.add(CommonConfig::new);
         ConfigManager.add(ClientConfig::new);
         ConfigManager.initialize();
 
-        RegistryUtils.begin();
-
         ItemTags.initialize();
         Items.initialize();
         Containers.initialize();
 
-        DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> ScanResultProviders::initialize);
-
-        RegistryUtils.finish();
-
-        FMLJavaModLoadingContext.get().getModEventBus().register(CommonSetup.class);
-        DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> FMLJavaModLoadingContext.get().getModEventBus().register(ClientSetup.class));
+        Network.initialize();
+        //DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> FMLJavaModLoadingContext.get().getModEventBus().register(ClientSetup.class));
```

Comment:
WIP or can this be removed?

Reply:
Removed; I commented it out while porting.